new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonUtils.java

```diff
@@ -0,0 +1,655 @@
+
+    // build the archive list
+    String zipFile = String.format("%s/%s.zip", ARCHIVE_DIR, appName);
+    final List<String> archiveList = Collections.singletonList(zipFile);
+
+    // Set additional environment variables for WIT
+    checkDirectory(WIT_BUILD_DIR);
+    Map<String, String> env = new HashMap<>();
+    env.put("WLSIMG_BLDDIR", WIT_BUILD_DIR);
+
+    // For k8s 1.16 support and as of May 6, 2020, we presently need a different JDK for these
+    // tests and for image tool. This is expected to no longer be necessary once JDK 11.0.8 or
+    // the next JDK 14 versions are released.
+    String witJavaHome = System.getenv("WIT_JAVA_HOME");
+    if (witJavaHome != null) {
+      env.put("JAVA_HOME", witJavaHome);
+    }
+
+    // build an image using WebLogic Image Tool
+    logger.info("Creating image {0} using model directory {1}", image, MODEL_DIR);
+    boolean result = createMiiImage(
+        defaultWitParams()
+            .modelImageName(imageName)
+            .modelImageTag(imageTag)
+            .modelFiles(modelList)
+            .modelArchiveFiles(archiveList)
+            .wdtVersion(WDT_VERSION)
+            .env(env)
+            .redirect(true));
+
+    assertTrue(result, String.format("Failed to create the image %s using WebLogic Image Tool", image));
+
+    // Check image exists using docker images | grep image tag.
+    assertTrue(doesImageExist(imageTag),
+        String.format("Image %s does not exist", image));
+
+    return image;
+  }
+
+  /**
+   * Create a Docker registry secret in the specified namespace.
+   *
+   * @param namespace the namespace in which the secret will be created
+   */
+  public static void createDockerRegistrySecret(String namespace) {
+
+    // Create Docker registry secret in the namespace to pull the image from repository
+    JsonObject dockerConfigJsonObject = createDockerConfigJson(
+        REPO_USERNAME, REPO_PASSWORD, REPO_EMAIL, REPO_REGISTRY);
+    String dockerConfigJson = dockerConfigJsonObject.toString();
+
+    // Create the V1Secret configuration
+    V1Secret repoSecret = new V1Secret()
+        .metadata(new V1ObjectMeta()
+            .name(REPO_SECRET_NAME)
+            .namespace(namespace))
+        .type("kubernetes.io/dockerconfigjson")
+        .putDataItem(".dockerconfigjson", dockerConfigJson.getBytes());
+
+    boolean secretCreated = assertDoesNotThrow(() -> createSecret(repoSecret),
+        String.format("createSecret failed for %s", REPO_SECRET_NAME));
+    assertTrue(secretCreated, String.format("createSecret failed while creating secret %s in namespace %s",
+        REPO_SECRET_NAME, namespace));
+  }
+
+  /**
+   * Docker login and push the image to Docker registry.
+   *
+   * @param dockerImage the Docker image to push to registry
+   */
+  public static void dockerLoginAndPushImageToRegistry(String dockerImage) {
+    // docker login, if necessary
+    if (!REPO_USERNAME.equals(REPO_DUMMY_VALUE)) {
```

Comment:
you really should be checking the registry name matches the dockerImage arg

Reply:
fixed